make/common/JavaCompilation.gmk

```diff
@@ -282,8 +282,23 @@ define SetupJavaCompilationBody
     $1_FLAGS += -Xlint:$$(call CommaList, $$(addprefix -, $$($1_DISABLED_WARNINGS)))
   endif
 
-  ifneq ($$($1_CLASSPATH), )
-    $1_FLAGS += -cp $$(call PathList, $$($1_CLASSPATH))
+  $1_AUGMENTED_CLASSPATH := $$($1_CLASSPATH)
+  $1_API_TARGET := $$($1_BIN)$$($1_MODULE_SUBDIR)/_the.$1_pubapi
+  $1_API_INTERNAL := $$($1_BIN)$$($1_MODULE_SUBDIR)/_the.$1_internalapi
+
+  ifeq ($$($1_CREATE_API_DIGEST), true)
+    $1_API_DIGEST_FLAGS := \
+        -Xplugin:"depend $$($1_API_TARGET)" \
+        "-XDinternalAPIPath=$$($1_API_INTERNAL)" \
+        "-XDLOG_LEVEL=$(LOG_LEVEL)" \
+        #
+
+    $1_EXTRA_DEPS := $$(BUILDTOOLS_OUTPUTDIR)/depend/_the.COMPILE_DEPEND_batch
+    $1_AUGMENTED_CLASSPATH += $$(BUILDTOOLS_OUTPUTDIR)/depend
```

Comment:
I think it would be better if you added `$$($1_BIN)` to the classpath whenever CREATE_API_DIGEST is true, rather than hard-coding it in the case of the microbenchmarks. 

Maybe you can also add a couple of lines of comments explaining the rationale, if you can make it succinct enough.